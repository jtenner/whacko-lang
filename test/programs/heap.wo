export fn _start(): void {
  let a = heap.malloc(usize(100));
  heap.free(a);
}