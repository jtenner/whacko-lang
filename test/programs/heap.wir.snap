program:
  callable:[std/builtins.wo]gc.initialize;functionType:()=>voidType
(extern function)
  callable:[std/builtins.wo]gc.alloc;functionType:(u32Type,usizeType)=>ptrType
(extern function)
  callable:[std/builtins.wo]gc.store_field_ptr;functionType:(i32Type,ptrType,ptrType,ptrType)=>voidType
(extern function)
  callable:[test/programs/heap.wo]_start;functionType:()=>voidType
    block: entry~470
      479~inst = (u32 100)
      481~inst = malloc: (runtime usizeType 479~inst)
      482~inst = ptrtoint: (runtime ptrType 481~inst)
      483~inst = store: (variable a) = (runtime usizeType 482~inst)
      484~inst = load: (variable a)
      486~inst = inttoptr: (runtime usizeType 484~inst) as ptrType
      free: (runtime ptrType 486~inst)
      return (void)

